### PR TITLE
fsp_srv: Add additional save data and content provider commands

### PR DESCRIPTION
**Pull request by DarkLordZach**
This adds various accessors that were made easy with the FileSystemController PR.

### CODE REVIEW

File: src/core/hle/service/filesystem/fsp_srv.cpp [hunk 1]
```diff
@@ -815,44 +981,103 @@ void FSP_SRV::CreateSaveDataFileSystem(Kernel::HLERequestContext& ctx) {
     rb.Push(RESULT_SUCCESS);
 }
 
-void FSP_SRV::OpenSaveDataFileSystem(Kernel::HLERequestContext& ctx) {
-    LOG_INFO(Service_FS, "called.");
+void FSP_SRV::CreateSaveDataFileSystemBySystemSaveDataId(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp{ctx};
+
+    auto save_struct = rp.PopRaw<FileSys::SaveDataDescriptor>();
```
Comment (lioncash) [inline src/core/hle/service/filesystem/fsp_srv.cpp]: This and the following variable can be const.

**Comment by bunnei** [comment]
@DarkLordZach reminder to address comments here

**Comment by bunnei** [comment]
Removing mainline tag because this will conflict with #3062, which I want to test

@DarkLordZach reminder - can you address comments here, so we can merge?

**Comment by momala454** [comment]
this pull cause crash in some games, see my proposed fix there https://github.com/yuzu-emu/yuzu/issues/3031

**Comment by lioncash** [comment]
Closing due to inactivity

**Pull request closed without merging**